sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/PersonalizerAsyncClient.java

```diff
@@ -0,0 +1,310 @@
+ *
+ * @see PersonalizerClientBuilder
+ */
+@ServiceClient(builder = PersonalizerClientBuilder.class, isAsync = true)
+public final class PersonalizerAsyncClient {
+
+    private final ClientLogger logger = new ClientLogger(PersonalizerAsyncClient.class);
+    private final PersonalizerClientV1Preview3Impl service;
+
+    PersonalizerAsyncClient(PersonalizerClientV1Preview3Impl service) {
+        this.service = service;
+    }
+
+    /**
+     * Request a list of actions to be ranked.
+     *
+     * <p>Submit a Personalizer rank request. Receives a context and a list of actions. Returns which of the provided
+     * actions should be used by your application, in rewardActionId.
+     *
+     * @param rankRequest A Personalizer Rank request.
+     * @return returns which action to use as rewardActionId, and additional information about each action as a result
+     * of a Rank request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PersonalizerRankResult> rank(PersonalizerRankOptions rankRequest) {
+        return rankWithResponse(rankRequest).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Request a list of actions to be ranked.
+     *
+     * <p>Submit a Personalizer rank request. Receives a context and a list of actions. Returns which of the provided
+     * actions should be used by your application, in rewardActionId.
+     *
+     * @param rankRequest A Personalizer Rank request.
+     * @return returns which action to use as rewardActionId, and additional information about each action as a result
+     * of a Rank request along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<PersonalizerRankResult>> rankWithResponse(PersonalizerRankOptions rankRequest) {
+        try {
+            return withContext(context -> rankWithResponse(rankRequest, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<PersonalizerRankResult>> rankWithResponse(PersonalizerRankOptions rankRequest, Context context) {
+        if (rankRequest == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'rankRequest' is required and cannot be null"));
+        }
+        return service.rankWithResponseAsync(rankRequest, context)
+            .onErrorMap(Transforms::mapToHttpResponseExceptionIfExists)
+            .map(response -> new SimpleResponse<>(response, response.getValue()));
+    }
+
+    /**
+     * Send a reward for an event.
+     *
+     * <p>Report reward between 0 and 1 that resulted from using the action specified in rewardActionId, for the
+     * specified event.
+     *
+     * @param eventId The event id this reward applies to.
+     * @param rewardValue The reward should be a floating point number, typically between 0 and 1.
+     * @return the {@link Response}.
```

Comment:
returns void, no response

Reply:
missing throws in javadoc here and below

Reply:
Fixed.